packages/hardhat-core/src/builtin-tasks/flatten.ts

```diff
@@ -115,6 +117,11 @@ subtask(
         flattened = `// SPDX-License-Identifier: ${license}\n\n${flattened}`;
       }
 
+      if (unifyABIEncoderV2) {
+        // Remove every line started with "pragma experimental ABIEncoderV2;" except the first one
+        flattened = flattened.replace(/pragma experimental ABIEncoderV2;\n/gm, (i => (m: string) => !i++ ? m : '')(0));
```

Comment:
This line is very hard to understand, is there a way to make it more readable?

Reply:
Ok, I just want to keep it as functional as I can. It was a time ago the last time I've used an IIFE.

Do you think that with better variables names works? 

```ts
flattened = flattened.replace(/pragma experimental ABIEncoderV2;\n/gm, (index => (match: string) => !index++ ? match: '')(0));
```

If not, alternative:

```ts
let index = 0
flattened = flattened.replace(/pragma experimental ABIEncoderV2;\n/gm, (match: string) => !index++ ? match : '');
```

Other solutions will include maybe a complex regex or the usage of `split` & `replace`.

Any other idea?

Reply:
I just pushed a change for this to make it more like the code we tend to write. We usually avoid compact one-liners and err on the "verbose but clear" side.